lib/json-schema/json-schema-resolver.ts

```diff
@@ -0,0 +1,51 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import util = require("util");
+
+export class JsonSchemaResolver implements IJsonSchemaResolver {
+	private schema: ISchema = null;
+
+	constructor(private schemas: ISchema[],
+				private $errors: IErrors) { }
+
+	public getSchema(schemaId: string): ISchema {
+		if(!this.schema) {
+			this.schema = this.findSchema(schemaId);
+
+			if(!this.schema) {
+				this.$errors.fail("Unable to find schema with id %s.", schemaId);
+			}
+
+			var extendsProperty = this.schema.extends;
+			if(extendsProperty) {
+				this.schema.extends = {};
+				this.schema.extends.properties = Object.create(null);
+
+				this.buildValidationSchema(extendsProperty);
+			}
+		}
+
+		return this.schema;
+	}
+
+	private buildValidationSchema(extendsProperty: ISchemaExtends[]) {
```

Comment:
missing return type (I'm not sure what is our convention, but I prefer to have them explicitly written)